app/models/account.rb

```diff
@@ -4,4 +4,6 @@ class Account < ApplicationRecord
   delegated_type :accountable, types: %w[ Account::Credit Account::Depository Account::Investment Account::Loan Account::OtherAsset Account::OtherLiability Account::Property Account::Vehicle], dependent: :destroy
 
   delegate :type_name, to: :accountable
+
+  monetize :balance, as: :balance_cents
```

Comment:
As mentioned by @rmarescu, I think long-term, we'll want to use the `money-rails` convention and avoid the `as:` argument so that we get a `balance_cents` and `balance_currency` field as DB columns.  This will apply to all monetary fields moving forward (accounts, transactions, etc.)

Can we write a migration to remove the existing `balance` column and replace with these two (using `add_monetize` helper)?

Reply:
Thank you for the feedback @zachgoll and @rmarescu .. Will implement suggested changes and push the code.

Reply:
```ruby
    ...
    Account.find_each do |account|
      account.update_columns(balance_cents: account.balance * 100)
    end
    ...
```

Multiplying the account balance by 100 is _mostly_ acceptable here since `balance` datatype is a bigint (wouldn't run into rounding issues).
However, this assumes that the subunit for the account's currency is always 100, which isn't the case for some currencies.
As an example, JPY's `subunit_to_unit` is 1:
https://github.com/RubyMoney/money/blob/092ecc7a711263208ae12fd62e7538f3d7a6348a/config/currency_iso.json#L1080-L1087

in which case we want to multiply the balance by 1 instead of 100.

I see that the default currency is 'USD' whose `subunit_to_unit` is 100, so the above code is fine.
But if we want to properly convert amounts to cents with respect to its currency, maybe we want something like:
```ruby
Account.find_each do |account|
  account.update_columns(balance_cents: Money.from_amount(account.balance, account.currency).cents)
end
```

Reply:
Thanks @loumarven I have updated migration code as per above suggestion. It will indeed make correct update in case of JPY currency.

Reply:
@zachgoll I have pushed suggested changes. Just added one more line to update balance_cents datatype to bigint